src/dashboard/src/middleware/common.py

```diff
@@ -65,3 +66,28 @@ def process_exception(self, request, exception):
             return
         if isinstance(exception, self.EXCEPTIONS):
             return render(request, 'elasticsearch_error.html', {'exception_type': str(type(exception))})
+
+
+SHIBBOLETH_REMOTE_USER_HEADER = getattr(
+    settings, 'SHIBBOLETH_REMOTE_USER_HEADER', 'REMOTE_USER'
+)
+
+
+class CustomShibbolethRemoteUserMiddleware(ShibbolethRemoteUserMiddleware):
+    """
+    Custom version of Shibboleth remote user middleware
+
+    THe aim of this is to provide a custom header name that is expected
+    to identify the remote Shibboleth user
+    """
+    header = SHIBBOLETH_REMOTE_USER_HEADER
+
+    def make_profile(self, user, shib_meta):
```

Comment:
`make_profile` is a hook from django-shibboleth-remoteuser which is called when a user authenticates